pyface/ui/wx/grid/grid.py

```diff
@@ -347,7 +347,16 @@ def dispose(self):
             window.Unbind(wx.EVT_LEFT_DOWN)
             window.Unbind(wx.EVT_LEFT_UP)
 
+        smobs = self.model.observe
         obs = self.observe
+        smobs(self._on_model_content_changed, "content_changed", remove=True)
+        smobs(
+            self._on_model_structure_changed, "structure_changed", remove=True
+        )
+        smobs(self._on_row_sort, "row_sorted", remove=True)
+        smobs(self._on_column_sort, "column_sorted", remove=True)
```

Comment:
although these are observers on the model not the grid itself I assume they should still be removed when the grid is disposed (notably because the handlers are methods on a mid disposal grid object)